### PR TITLE
TEST/ASYNC: Drain pipe before test

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by Artemy-Mellanox** [comment]
pipe behavior changes in upstream 5.5
https://github.com/torvalds/linux/commit/f467a6a66419a18f782f87507464a5dedc942dec

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 05-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/f1ca8d1b1a2b924c381edda764ee1cceb0d63eca/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/954ef1a072d17dabc03907bfb11dbac6b46e358b/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/d0d7097b1811dbe5ca4cc31b4b07b994d1aaa5ff/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/0233d0e6af34f45cab502d701aa937880138e202/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/7bb95b6f12c836f84d8a729e21c26461a007df4f/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/76d9408186edd0b9dd6424f683f0dd34aece39e4/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/a7bab89bdd33ff089b681aa9324635c8298a539f/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/ded62a04ef624e615e1c2bca04c753d203dfbe62/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/3c79b00606eacac2069b97d92789d985405dec79/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/3d37e79163efeaaf628dea65712432ad16ceb37e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/3b55a135d41ef90e0f54cfcaaa8d1413b203540b/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/282d5851c0e4e101a4c054b1b92374f07de8f48c/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/0c89b63774e877a8a0cc9ce5ba7a1039eb2e92e4/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/9fa54ee3edfda23620ff3b5768d58028e8a73382/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/af0186ae44130d8a897babc0672e56699dc12b55/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/401d4724a134b24a47a549eb45614618e4ea16eb/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/31ae68bcf1b92ea3e3f6ca06f0dc06398109e1d9/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/8196a7a470b3c0b7d9cb04e1ed30c4bacd124f03/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/54701e7a340a796bda3e546cff62fa18026f3530/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/62f145957b70d993ba7c60123985c9d83935fca5/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/9738175266875e549f66849ef5161ab22b4d8b60/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/06babad370d477989ed04660b5d8abfce7ef0505/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/273f8a53d70e5b2109ebeb114c4cf83f52816f86/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/6c13ceb4be73b248993ccf25ddd30c0cf3f32b2f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/417e4fa2a2cc34d07e783727c831e872/raw/8c6ee8ad69930542eec84ca10fb1250ca96d2c83/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by Artemy-Mellanox** [comment]
@yosefe

File: test/gtest/ucs/test_async.cc [hunk 1]
```diff
@@ -576,6 +580,7 @@ UCS_TEST_P(test_async, modify_event) {
     le.push_event();
     suspend_and_poll(&le, COUNT);
     EXPECT_EQ(le.count(), count);
+    le.drop_event();
```
Comment (yosefe) [inline test/gtest/ucs/test_async.cc]:
```
diff --git a/test/gtest/ucs/test_async.cc b/test/gtest/ucs/test_async.cc
index bbf49b870..3befe12a1 100644
--- a/test/gtest/ucs/test_async.cc
+++ b/test/gtest/ucs/test_async.cc
@@ -95,9 +95,13 @@ public:
         ucs_async_pipe_push(&m_event_pipe);
     }
 
+    void reset() {
+        ucs_async_pipe_drain(&m_event_pipe);
+    }
+
 protected:
     virtual void ack_event() {
-        ucs_async_pipe_drain(&m_event_pipe);
+        reset();
     }
 
 private:
@@ -576,6 +580,7 @@ UCS_TEST_P(test_async, modify_event) {
     le.push_event();
     suspend_and_poll(&le, COUNT);
     EXPECT_EQ(le.count(), count);
+    le.reset();
 
     ucs_async_modify_handler(le.event_id(), UCS_EVENT_SET_EVREAD);
     count = le.count();
```

File: test/gtest/ucs/test_async.cc [hunk 2]
```diff
@@ -95,6 +95,10 @@ class base_event : public base {
         ucs_async_pipe_push(&m_event_pipe);
     }
 
+    void reset() {
+        ack_event();
```
Comment (yosefe) [inline test/gtest/ucs/test_async.cc]: ack_event should call reset, not other way

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 4 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 05-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/22b4baf907eb66c62282009b8a6b029023633e9f/_main) | :question: ABORTED
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/02a1fc5b77cd80ce4f2e67f71bc3017708b0e50b/hpc-arm-hwi-jenkins_W1) | :question: ABORTED
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/fc4e1525873091a52cb96f06e6e4bfbd3f2a1782/hpc-arm-hwi-jenkins_W3) | :question: ABORTED
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/81f84aa404e67975391583151937ffc49030fc0c/hpc-test-node-gpu_W0) | :question: ABORTED
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/c519b8f6537e96b49e94b63c262f286e3d891907/hpc-test-node-gpu_W1) | :question: ABORTED
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/a2c8cc75b9c279c1ea35da678db3c1f93fb82555/hpc-test-node-gpu_W2) | :question: ABORTED
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/791bc0b97964232781658d6a458322a4b3e1e8f5/hpc-test-node-gpu_W3) | :question: ABORTED
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/34481017d12bf65d4f7fcb1ca81478374e706fc8/hpc-test-node-legacy_W0) | :question: ABORTED
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/23e4264c09e07218f92b3f3fef846b452e9efbbb/hpc-test-node-legacy_W1) | :question: ABORTED
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/af3f5ee44f8c59135ba5e02e35a04fae8c072b20/hpc-test-node-legacy_W2) | :question: ABORTED
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/611a9091b2fa98c1033afc3f337a083208f04fb3/hpc-test-node-legacy_W3) | :question: ABORTED
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/698496204ea1dc09647a8f9f527da34ee17f42d7/hpc-test-node-new_W0) | :question: ABORTED
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/6d2b3b456155821682a2fc9c6a708532a8d2052f/hpc-test-node-new_W1) | :question: ABORTED
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/0d1750781e36e98728913cd4a9e6cadb80722bdd/hpc-test-node-new_W2) | :question: ABORTED
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/ee8c3fe478a253968f24c042f38946144b8325d1/hpc-test-node-new_W3) | :question: ABORTED
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/1a4c7eff412cb48d9253d1fe417740a887603c54/hpc-arm-cavium-jenkins_W0) | :x: FAILURE
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/6f809cdc6b56381a7387dfcf5d7b92554f6008e7/hpc-arm-cavium-jenkins_W1) | :x: FAILURE
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/b7679ed662482958c2aacd3c51143a4878c0667f/hpc-arm-cavium-jenkins_W2) | :x: FAILURE
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/a8e335acab115eae4d80b287f2a53fd06e0da430/hpc-arm-cavium-jenkins_W3) | :x: FAILURE
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/e83fb98e1aa02d2e60a6d4c14ad6f4148e80f399/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/9e7c8fbf0b97c5ac7bbc4dcd6d77570442922402/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/3ca9bd73263511deff380008d4578b1f730b582e/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/f683a4a541f553a43ef15a461f98f970ab73e2ee/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/5b0f7594c0c63a2c18c2bcc58a4eb6523f398f23/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/49686effbab955ab72123fbd971aa53a/raw/0d5652f45a6237039c7d33f29480e6581f76425b/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 05-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/466a72a6fdc0a26ed229dca84ba2b87a3ae542fc/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/7714d2bf2e63f3bde7d03a25fb732eae91f50143/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/b2b65d6fb11cd166f2e3fb474af6af10fc84fe43/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/0267071ae556f08d17b49ed7cd280fd062793d08/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/e0fa62d05b31de84de044469dbdc66ce3d45c864/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/8381c237ac2a80fec050bba9117658f549c745ee/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/8819558d9ae5f995ffcf773bf3229d443653f6a1/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/a11aea7cb4a401f79765ace66e2d61c5b90749ea/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/67c6e138804f167c0403f6c39acbf9071fe939ff/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/8d6233d2d7bbd6af9d9289ded2cc50ede9acfd82/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/f79fae94edfc38f55088bc641d4d51b3f152b1db/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/019c50cbd13364483e79fcdcdd268f6c85d6dc6b/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/6736cb2b90f837cdf063d4c92e418d812ca0bb54/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/facae4ee8fe38b8ca52dfbc390b0ad2b7fb5d65b/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/df0ebcf072812b0558f9f1eacea3ab54030ac6d3/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/25116a9dbf9cd6602df3e547a2b33fe0824258fd/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/830d205b91ce96abfdea243d225445cc38cf41dc/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/714f5367adc995a375282a0e5b4bd98001f129f5/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/fd21c4562171e37c08ae3006ba2c93aa2a5f6864/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/5fee30203d616eeb1ca5186c70da22292a002e6f/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/253cd440c8f62ef71de2750adf80357d763c341a/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/c330282aa94ec417a0fb515e4daf36882508e0bd/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/d7907cedcb5814369390be79b2af0d2d353b05d2/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/8c3b5ec41571157559f72b3e85275c5007a6b70b/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ec5eb9484647d1e92202f4e477cd2216/raw/92f538a46c03b373415194a430fe3c9fac9632d4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**